### ISSUE TITLE
need for subsetting regions for backtracking

### ISSUE THREAD

**Issue by eabarnes1010** [WAM2layers/WAM2layers#252]
As it stands, the backtracking code will run over the entire domain of the era5 data. However, one should not have to re-download (or resave) the ERA5 data for each separate domain they may want to perform the backtracking on. In my view, the best way to use the WAM for future flexibility is to download the global ERA5 data and run the pre-processing (fluxes/states) on that data. Then, at the backtracking stage, have the user specify the domain over which to perform the backtracking. This could be a subset of the globe, or the entire globe. I have written code to do this, but am putting in an issue to start discussion.

**Comment by Peter9192** [comment]
Thanks for opening the issue! Linking your draft PR just to help in the discussion: #251.

I agree the way you describe sounds like a good way to conduct many experiments, and it's a useful feature. With respect to how to facilitate that, we have (at least) two options:

1 Similar to how we use `source_region.nc`: use global pre-processed data but create a seperate `track-region.nc` file with smaller extents. Advantage: allows for flexible or even time-dependent boundaries (do we ever need that?) and is consistent with source region definition. Disadvantage: you need to make a separate nc file to modify the tracking region
2. Add an additional config argument to set a bounding box, as proposed in @251. Advantage: simple to change, intuitive. Disadvantage: less flexible

Please add any thoughts

**Comment by Peter9192** [comment]
With the bbox, it would be good to stick with a default convention for the order of lat/lon min/max. E.g., we could use what [geopandas also uses](https://geopandas.org/en/stable/docs/reference/api/geopandas.read_file.html#geopandas-read-file): `minx, miny, maxx, maxy` in our case `west, south, east, north`. Are there any other conventions out there?

**Comment by eabarnes1010** [comment]
Good call about the default convention, yes that should done. 

Also, I like your thinking about a separate track-region.nc. Honestly, I feel like that could be even cleaner. A disadvante though: it is more work to change your tracking region than a few keystrokes of specifying the new tracking. But I don't really see this as something someone will be looping through the parameter space for often, so I think I generally like your idea better (and keeps consistent with source_region.nc). 

Glad we are going the discussion route and not my immediate PR "proposal". Thank you for teaching me about this!

**Comment by Peter9192** [comment]
After giving it some more thought, I wonder if we could support both options. It could look like this in your config:

```py
    reduced_domain_bounds: Union[FilePath, BoundingBox]
    """(Reduced) domain to perform the backtracking

    You can either specify a path that contains a 
    netcdf file, or a bounding box of the form [..., ..., ..., ...].
    ...
    """
```

The bounding box type could be specified e.g. like so (see https://docs.pydantic.dev/dev-v2/usage/types/list_types/#namedtuple):

```py
class BoundingBox(NamedTuple):
    west: float
    south: float
    east: float
    north: float
```

We could do the same for the source region. 

What about the name for this new config parameter? Perhaps we can align it with the source region, like `source_region` vs `tracking_region`?

**Comment by eabarnes1010** [comment]
I think all of this sounds great. I like the consistency with what you already have!

**Comment by Peter9192** [comment]
Cool! So how to proceed? Do you think you could work a bit more on your PR to align with what we've discussed above, and/or do you need some help with that?

**Comment by eabarnes1010** [comment]
I think that help would be much appreciated. One of my concerns is that I don't fully understand how you and Ruud etc use this with data from a restricted domain (I would always use the entire ERA5 global domain myself) so I think your help would be great.

**Comment by Peter9192** [comment]
I guess we'd have to do a check somewhere to see whether the preprocessed data fully contains the subset provided by the user. I'll try to find some time to take a closer look, but it might take a while before I get to it.

**Comment by Peter9192** [comment]
This has been implemented in #297. Feel free to reopen or open a new issue if you have any further thoughts or if this doesn't work for you.
